packages/deployment-service-ui/src/components/pages/pipelines/list.tsx

```diff
@@ -78,85 +193,13 @@ export default () => {
-                      {
-                        label: 'Name',
-                        value: pipeline.name as string,
-                      },
-                      {
-                        label: 'Domain',
-                        value: pipeline.subDomain as string,
-                      },
-                      {
-                        label: 'repository',
-                        value: pipeline.repository as string,
-                      },
-                      {
-                        label: '',
-                        value: '',
-                        children: (
-                          <Button
-                            intent="secondary"
-                            onClick={() => {
-                              history.push(Routes.PIPELINES_SHOW.replace(':pipelineId', pipeline.id as string))
-                            }}
-                          >
-                            Manage
-                          </Button>
-                        ),
-                      },
-                    ],
-                  }))}
-                />
-                {pipelinePagination.items.length ? (
-                  <PaginationWrap>
-                    <PaginationText>
-                      <strong>{pipelinePagination.meta.currentPage}</strong> of {pipelinePagination?.meta.totalPages}
-                    </PaginationText>
-                    <PaginationButton
-                      onClick={async () => {
-                        if (pipelinePagination.meta.currentPage <= 1) {
-                          return
-                        }
-                        await fetchPipelines(pipelinePagination.meta.currentPage - 1)
-                      }}
-                    >
-                      <Icon icon="backSystem" />
-                    </PaginationButton>
-                    <PaginationButton
-                      onClick={async () => {
-                        if (pipelinePagination.meta.currentPage >= pipelinePagination.meta.totalPages) {
-                          return
-                        }
-                        await fetchPipelines(pipelinePagination.meta.currentPage + 1)
-                      }}
-                    >
-                      <Icon icon="nextSystem" className={elPaginationPrimary} />
-                    </PaginationButton>
-                  </PaginationWrap>
-                ) : (
-                  <PersistantNotification intent="secondary" isExpanded isInline isFullWidth>
-                    No pipelines retrieved. You will need to create a new pipeline from the left hand side menu.
-                  </PersistantNotification>
-                )}
-              </section>
+              <PipelineList
+                setPipelinePagination={setPipelinePagination}
+                connection={connectSession as ReapitConnectSession}
+                pipelinePagination={pipelinePagination}
+                fetchPipelines={fetchPipelines}
+                history={history}
```

Comment:
I think you could use the `useHistory` react-router-dom hook instead of passing history around?

Reply:
yea good point!